File: backend/interview_portal/src/main/java/by/interview/portal/dto/InterviewDTO.java
package by.interview.portal.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class InterviewDTO {

    private Long id;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private CandidateBaseInfoDTO candidate;

    private String place;

    private String status;

    private Set<UserBaseInfoDTO> interviewerSet;

    private Set<DisciplineBaseInfoDTO> disciplineSet;

}

File: frontend/src/app/api/models/interview-dto.ts
/* tslint:disable */
import { CandidateBaseInfoDTO } from './candidate-base-info-dto';
import { DisciplineBaseInfoDTO } from './discipline-base-info-dto';
import { UserBaseInfoDTO } from './user-base-info-dto';

export interface InterviewDTO {

  candidate?: CandidateBaseInfoDTO;

  disciplineSet?: DisciplineBaseInfoDTO[];

  endTime?: string;

  id?: number;

  interviewerSet?: UserBaseInfoDTO[];

  place?: string;

  startTime?: string;

  status?: string;
}
